packages/mobile/src/components/audio/Audio.tsx

```diff
@@ -419,35 +317,49 @@ export const Audio = () => {
     }
   }
 
-  return (
-    <View style={styles.backgroundVideo}>
-      {source && (
-        <Video
-          // @ts-ignore: type: m3u8 is actually a valid prop override
-          source={source}
-          ref={videoRef}
-          playInBackground
-          playWhenInactive
-          allowsExternalPlayback={false}
-          audioOnly
-          // Mute playback if we are casting to an external source
-          muted={isCasting}
-          onError={handleError}
-          onEnd={() => {
-            setDuration(0)
-            pause()
-            onNext()
-          }}
-          progressUpdateInterval={100}
-          onLoad={(payload) => {
-            setDuration(payload.duration)
-          }}
-          onProgress={onProgress}
-          repeat={repeatMode === RepeatMode.SINGLE}
-          paused={!playing}
-          // onBuffer={this.onBuffer}
-        />
-      )}
-    </View>
+  const currentUriRef = useRef<string | null>(null)
+
+  const handleSourceChange = useCallback(
+    async (newUri: string) => {
+      const currentUri = currentUriRef.current
+
+      if (currentUri !== newUri) {
+        currentUriRef.current = newUri
+        const imageUrl = trackImageSource?.source[2].uri ?? DEFAULT_IMAGE_URL
+        await TrackPlayer.reset()
```

Comment:
So is this the jank part where we need to reset track player whenever starting new track?

Reply:
Yea exactly, stable enough that it works without any hitches, but not the intended way to do it for sure